Code/Source/RobotImporter/URDF/PrefabMakerUtils.cpp

```diff
@@ -0,0 +1,134 @@
+
+        transformInterface->SetLocalTM(tf);
+    }
+
+    AzToolsFramework::Prefab::PrefabEntityResult CreateEntity(AZ::EntityId parentEntityId, const AZStd::string& name)
+    {
+        auto prefabInterface = AZ::Interface<AzToolsFramework::Prefab::PrefabPublicInterface>::Get();
+        auto createEntityResult = prefabInterface->CreateEntity(parentEntityId, AZ::Vector3());
+        if (!createEntityResult.IsSuccess())
+        {
+            return createEntityResult;
+        }
+
+        AZ::EntityId entityId = createEntityResult.GetValue();
+        if (!entityId.IsValid())
+        { // Verify that a valid entity is created.
+            return AZ::Failure(AZStd::string("Invalid id for created entity"));
+        }
+
+        AZ_TracePrintf("CreateEntity", "Processing entity id:%s with name:%s", entityId.ToString().c_str(), name.c_str());
+        AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
+        AZStd::string entityName(name.c_str());
+        entity->SetName(entityName);
+        entity->Deactivate();
+        AddRequiredComponentsToEntity(entityId);
+        return createEntityResult;
+    }
+
+    AzToolsFramework::Prefab::PrefabOperationResult RemoveEntityWithDescendants(AZ::EntityId parentEntityId)
+    {
+        auto prefabInterface = AZ::Interface<AzToolsFramework::Prefab::PrefabPublicInterface>::Get();
+        return prefabInterface->DeleteEntitiesAndAllDescendantsInInstance({ parentEntityId });
+    }
+
+    void AddRequiredComponentsToEntity(AZ::EntityId entityId)
+    {
+        AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
+        AzToolsFramework::EditorEntityContextRequestBus::Broadcast(
+            &AzToolsFramework::EditorEntityContextRequests::AddRequiredComponents, *entity);
+    }
+
+    bool HasCollider(AZ::EntityId entityId)
+    { // TODO - actually, we want EditorColliderComponent specifically, but the change can be applied only after moving to ECC
+        // TODO - which will happen when Cylinder shape is supported. Until then, we check for either ECC or ESCC.
+        AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
+        return entity->FindComponent<PhysX::EditorColliderComponent>() != nullptr ||
+            entity->FindComponent<PhysX::EditorShapeColliderComponent>() != nullptr;
+    }
+
+    AzToolsFramework::EntityIdList GetColliderChildren(AZ::EntityId parentEntityId)
+    {
+        AzToolsFramework::EntityIdList colliderChildren;
+        AzToolsFramework::EntityIdList allChildren = AzToolsFramework::GetEntityChildOrder(parentEntityId);
+        for (auto childId : allChildren)
+        {
+            AZ_TracePrintf("GetColliderChildren", "Considering child %s", childId.ToString().c_str());
+            if (HasCollider(childId))
+            {
+                AZ_TracePrintf("GetColliderChildren", "Child %s has a collider", childId.ToString().c_str());
```

Comment:
One thing that isn't known by users of `AZ_TracePrintf` is that unlike `AZ_Error`, `AZ_Warning` and `AZ_Assert` it doesn't automatically append a newline to the end of the message.

So the all the trace prtinfs will be output on a single line.

That being said, are these traceprints meant to stick around, since they seem like debug printfs to test this function and not to log useful user facing information.

Reply:
Thanks for that tip!